tests/spawn-after-drop.rs

```diff
@@ -0,0 +1,52 @@
+use std::{env, process::Command};
+
+use jobserver::{Client, FromEnvErrorKind};
+
+macro_rules! t {
+    ($e:expr) => {
+        match $e {
+            Ok(e) => e,
+            Err(e) => panic!("{} failed with {}", stringify!($e), e),
+        }
+    };
+}
+
+fn main() {
+    match env::args().skip(1).next().unwrap_or_default().as_str() {
+        "" => {
+            let me = t!(env::current_exe());
+            let mut cmd = Command::new(me);
+            let client = t!(Client::new(1));
+            client.configure(&mut cmd);
+            drop(client);
+            let status = t!(cmd.arg("from_env").status());
+            assert!(status.success());
+        }
+        "from_env" => {
+            let me = t!(env::current_exe());
+            let mut cmd = Command::new(me);
+            let client = unsafe {
+                t!(match Client::from_env_ext(true).client {
+                    // Its ok for a dropped jobservers path to no longer exist (e.g. on Windows).
+                    Err(e) if matches!(e.kind(), FromEnvErrorKind::CannotOpenPath) => return,
```

Comment:
IMHO it's kind of a bad behavior that it has different behavior on different platform.

Reply:
@ChrisDenton, one more case to start supporting this.

Reply:
I am happy to collaborate would @beetrees on this to stop processes from inheriting handles.

Reply:
Check out https://github.com/rust-lang/rust/pull/115501.